spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/UniProtKBDocumentsToHPSWriter.java

```diff
@@ -108,6 +111,23 @@ public void writeIndexDocumentsToHPS() {
         log.info("Completed UniProtKB prepare Solr index");
     }
 
+    /**
+     * @return RDD of JavaPairRDD<accesion, UniProtDocument> with inactive entries information
+     */
+    JavaPairRDD<String, UniProtDocument> getInactiveEntryRDD() {
+        UniParcRDDTupleReader uniParcRDDTupleReader = new UniParcRDDTupleReader(parameter, false);
+        JavaPairRDD<String, String> inactiveUniParc =
+                uniParcRDDTupleReader.load().flatMapToPair(new UniParcInactiveUniProtKBMapper());
+
+        JavaPairRDD<String, UniProtDocument> inactiveEntryRDD =
+                InactiveUniProtKBRDDTupleReader.load(parameter)
+                        .leftOuterJoin(inactiveUniParc)
+                        .mapValues(new UniParcDeletedUniProtKBJoin())
+                        .mapValues(new UniProtEntryToSolrDocument(Collections.emptyMap()));
+
+        return inactiveEntryRDD.repartition(inactiveEntryRDD.getNumPartitions() / 8);
```

Comment:
Whats the reason for repartition?

Reply:
It was crashing. It crashed 3 times, and I think that it was because we were saving more than 90K partitions in the filesystem just for the inactive entries. + around 60K for active entries, so in total it was around 150K files saved in the disk.
This repartition will reduce the number of partitions for inactive entries in the file system, and it will keep around the same value that it was before this change. Let me know if you need more details, We can do a call and I can give you more details.